### PR TITLE
557 import tool usability

### LINKED ISSUE #557: Import tool usability
1. Error messages should be more meaningful.
2. Required columns should be aligned between Parquet and ORC formats.
3. Importing the same inventory twice should fail immediately instead of starting the diff.
4. When importing a newer inventory, the progress bar/messages should indicate that a diff is occurring

### PR DESCRIPTION
**Pull request by johnnyaug**
Implements #557 

1. Align inventory column requirements between Parquet in ORC: make all columns except bucket and key optional in Parquet.
2. Add tests to parquet reader

### CODE REVIEW

File: cloud/aws/s3inventory/parquet_reader.go [hunk 1]
```diff
@@ -1,20 +1,127 @@
...
+		isDeleteMarker, err = cast.ToBoolE(v)
+		o.IsDeleteMarker = swag.Bool(isDeleteMarker)
+	case sizeFieldName:
+		var size int64
+		size, err = cast.ToInt64E(v)
+		o.Size = swag.Int64(size)
+	case lastModifiedDateFieldName:
+		var lastModifiedMillis int64
+		lastModifiedMillis, err = cast.ToInt64E(v)
+		o.LastModifiedMillis = swag.Int64(lastModifiedMillis)
```
Comment (arielshaqed) [inline cloud/aws/s3inventory/parquet_reader.go:106]: As before, would still be happier to have a proper `time.Time` here.

File: cloud/aws/s3inventory/reader_test.go [hunk 2]
```diff
@@ -119,82 +177,180 @@ func TestInventoryReader(t *testing.T) {
...
 		},
-		{
+		"orc with 12500 objects": {
 			ObjectNum:           12500,
 			ExpectedReadObjects: 12500,
 			ExpectedMinValue:    "f00000",
 			ExpectedMaxValue:    "f12499",
+			Format:              "ORC",
+		},
+		"orc with 100 objects": {
```
Comment (arielshaqed) [inline cloud/aws/s3inventory/reader_test.go:255]: Could we just run each of the test cases twice, once with each `Format`?

**Review by arielshaqed — APPROVED** [review]
Thanks!

Very nice, much cleaner.  Please go over comments but _this is good_ and *approved (almost) any way you handle comments*

File: cloud/aws/s3inventory/orc_reader.go [hunk 3]
```diff
@@ -2,7 +2,6 @@ package s3inventory
 
 import (
 	"context"
-	"reflect"
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply cloud/aws/s3inventory/orc_reader.go:5]: you were in my heart when I did it

File: cloud/aws/s3inventory/parquet_reader.go [hunk 4]
```diff
@@ -18,3 +27,87 @@ func (p *ParquetInventoryFileReader) FirstObjectKey() string {
...
+	return res, nil
+}
+
+func set(o *InventoryObject, f string, v interface{}) error {
+	var err error
+	switch f {
+	case "bucket":
+		var bucket string
+		bucket, err = cast.ToStringE(v)
+		o.Bucket = bucket
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply cloud/aws/s3inventory/parquet_reader.go:86]: :+1:

File: block/s3/inventory_iterator.go [hunk 5]
```diff
@@ -94,8 +94,7 @@ func (it *InventoryIterator) fillBuffer() bool {
 			it.logger.Errorf("failed to close manifest file reader. file=%s, err=%w", it.Manifest.Files[it.inventoryFileIndex].Key, err)
 		}
 	}()
-	it.buffer = make([]s3inventory.InventoryObject, rdr.GetNumRows())
-	err = rdr.Read(&it.buffer)
+	it.buffer, err = rdr.Read(int(rdr.GetNumRows()))
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply block/s3/inventory_iterator.go:97]: this will cause worse casting mess

File: cloud/aws/s3inventory/parquet_reader.go [hunk 6]
```diff
@@ -2,110 +2,115 @@ package s3inventory
...
+		o.Size, err = cast.ToInt64E(v)
+	case lastModifiedDateFieldName:
 		var lastModifiedMillis int64
 		lastModifiedMillis, err = cast.ToInt64E(v)
-		o.LastModifiedMillis = swag.Int64(lastModifiedMillis)
-	case "e_tag":
-		var checksum string
-		checksum, err = cast.ToStringE(v)
-		o.Checksum = swag.String(checksum)
+		o.LastModified = swag.Time(time.Unix(lastModifiedMillis/int64(time.Second/time.Millisecond), 0))
```
Comment (arielshaqed) [inline cloud/aws/s3inventory/parquet_reader.go:97]: Do we care about losing the subsecond resolution?
```suggestion
		o.LastModified = swag.Time(time.Unix(lastModifiedMillis/int64(1000), (lastModifiedMillis % 1000) * int64(1_000_000))
```

(also you can see that I'm not a fan of pretending that the number of milliseconds per second may change, and I'm trying to sneak that change by you...).

**Review by arielshaqed — APPROVED** [review]
Cool; thanks!  StilLGTM...

File: cloud/aws/s3inventory/parquet_reader.go (hunk 6 above)
Comment (johnnyaug) [reply cloud/aws/s3inventory/parquet_reader.go:97]: Our linter would never allow it !

**Pull request merged**